src/web/mjs/engine/DiscordPresence.mjs

```diff
@@ -0,0 +1,167 @@
+
+        // Current status
+        this.status = {
+            largeImageKey: 'logo',
+            largeImageText: 'Manga & Anime Downloader for Linux, Windows & MacOS'
+        };
+        this.statusNew = true;
+
+        // EvetnListener for:
+        // Connector selected
+        document.addEventListener( EventListener.onSelectConnector, this._onSelectConnector.bind(this) );
+        // Manga selected
+        document.addEventListener( EventListener.onMangaClicked, this._onMangaClicked.bind(this) );
+        // get chapter pages (Reader/Downloader)
+        document.addEventListener( EventListener.onChapterGetPages, this._onChapterGetPages.bind(this) );
+
+        this.startDiscordPresence();
+    }
+
+    _onSettingsChanged() {
+        switch (this._settings.discordPresence.value) {
+            case 'none':
+                this.enabled = false;
+                this.enabledHentai = false;
+                break;
+            case 'nohentai':
+                this.enabled = true;
+                this.enabledHentai = false;
+                break;
+            case 'hentai':
+                this.enabled = true;
+                this.enabledHentai = true;
+                break;
+            default:
+                this.enabled = false;
+                this.enabledHentai = false;
+        }
+
+        if (this.enabled) {
+            this.statusNew = true;
+            this.startDiscordPresence();
+        } else {
+            this.stopDiscordPresence();
+        }
+    }
+
+    _onSelectConnector(e) {
+        // Hentai check
+        if(e.detail.tags.includes('hentai') || e.detail.tags.includes('porn')) {
+            this.hentai = true;
+        } else {
+            this.hentai = false;
+        }
+
+        this.status['details'] = 'Browsing ' + e.detail.label;
+        if (this.status.state) delete this.status.state;
+        this.status.startTimestamp = + new Date();
+        this.statusNew = true;
+        if (this.enabled && !this.rpc) this.startDiscordPresence();
+    }
+
+    _onMangaClicked(e) {
+        // Hentai check
+        if(e.detail.connector.tags.includes('hentai') || e.detail.connector.tags.includes('porn')) {
+            this.hentai = true;
+        } else {
+            this.hentai = false;
+        }
+
+        this.status['details'] = 'Browsing ' + e.detail.connector.label;
+        this.status['state'] = 'Looking at ' + e.detail.title;
+        this.status.startTimestamp = + new Date();
+        this.statusNew = true;
+        if (this.enabled && !this.rpc) this.startDiscordPresence();
+    }
+
+    _onChapterGetPages(e) {
+        // Hentai check
+        if(e.detail.manga.connector.tags.includes('hentai') || e.detail.manga.connector.tags.includes('porn')) {
+            this.hentai = true;
+        } else {
+            this.hentai = false;
+        }
+
+        this.status['details'] = 'Reading ' + e.detail.manga.title;
```

Comment:
It might also be an anime, where reading does not fit

Reply:
How about "Viewing", should match both cases :)